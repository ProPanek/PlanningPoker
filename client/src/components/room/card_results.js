import React, {Component} from 'react'
import {inject, observer} from "mobx-react";
import {withRouter} from "react-router-dom";
import Typography from "@material-ui/core/Typography";
import styled from "styled-components";

import {Card} from "@material-ui/core";


const Wrapper = styled(Card)`
  display: flex;
  flex-direction: row;
  margin: 0 auto;
    margin-top:10px;

  width: 90%;
  flex-wrap: wrap;
`;

const CardsWrapper = styled.div`
  &&{
  margin: 0 auto;
  width: 90%;
  margin-bottom:10px;
  display:flex;
  }
`;

const StyledCard = styled(Card)`
  &&{
    width:57px;
    height:89px;
    font-size: -webkit-xxx-large;
    line-height: 89px;
    margin: 5px;
    margin: 0 auto;
    background-color:${props => props.color || "#303F9F"};
    color:white;
  }
`;


class CardResults extends Component {

  render() {

    const {store: {room: {waiting, cardResults}}} = this.props
    return (
      <React.Fragment>
        <Wrapper>
          <CardsWrapper>
            {waiting.length > 0 && waiting.map((result, index) => (
              <Card key={index} style={{width: "100px", minHeight: "20px"}}>
                <Typography>
                </Typography>
                <StyledCard color={"#303F9F"}>
                </StyledCard>
              </Card>
            ))}

          </CardsWrapper>
          <CardsWrapper>
            {cardResults.length > 0 && cardResults.map((result, index) => (
              <Card key={index} style={{width: "100px", minHeight: "20px"}}>
                <Typography>
                  {result.userName}
                </Typography>
                <StyledCard color={result.color}>
                  {result.cardValue}
                </StyledCard>
              </Card>
            )) || <Typography style={{margin: "0 auto"}}> Waiting for response from all users </Typography>}
          </CardsWrapper>
        </Wrapper>
      </React.Fragment>
    );
  }
}

export default inject("store")(withRouter(observer(CardResults)));

