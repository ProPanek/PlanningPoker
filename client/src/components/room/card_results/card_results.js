import React, {Component} from "react"
import {inject, observer} from "mobx-react";
import Typography from "@material-ui/core/Typography";
import Tooltip from '@material-ui/core/Tooltip';
import styled from "styled-components";
import Card from "@material-ui/core/Card";
import Zoom from '@material-ui/core/Zoom';
import ReactCardFlip from 'react-card-flip';
import {decorate, observable, reaction} from "mobx";


const Wrapper = styled(Card)`
  display: flex;
  flex-direction: row;
  margin: 10px auto 0;
  width: 90%;
  flex-wrap: wrap;
`;

const CardsWrapper = styled.div`
  &&{
  display:flex;
  width: 90%;
  margin: 20px auto 10px;
  flex-wrap: wrap;
  }
`;

const Wrap = styled.div`
  width: 100px;
  min-height: 20px;
  perspective: 1000px;
`

const StyledCard = styled(Card)`
  &&{
    width:57px;
    height:89px;
    font-size: 30px;
    line-height: 89px;
    margin: 5px;
    margin: 0 auto;
    background-color:${props => props.color || "#303F9F"};
    color:white;
  }
`;

const StyledResultCard = styled(Card)`
  &&{
    margin-right: 20px;
  }
`

class CardResults extends Component {
  constructor(props) {
    super(props)
    this.isFlipped = []
    reaction(
      () => this.props.store.roomStore.cardResults.length,
      () => {
        if (this.props.store.roomStore.cardResults.length > 0) {
          this.startFlipTimer()
        }
      }
    )
  }

  startFlipTimer = () => {
    const {store: {roomStore: {cardResults}}} = this.props
    cardResults.forEach((data, index) => {
      this.isFlipped[index] = false
      setTimeout(() => {
        this.isFlipped[index] = true
      }, 1000 * (index + 1))
    })
  }

  render() {
    const {store: {roomStore: {waiting, cardResults}}} = this.props
    return (
      <React.Fragment>
        <Wrapper>
          <CardsWrapper>
            {waiting.length > 0 && waiting.map((result, index) => (
              <Card key={index}>
                <StyledCard color={"#303F9F"}>
                </StyledCard>
              </Card>
            ))}
          </CardsWrapper>
          <CardsWrapper>
            {cardResults.length > 0 && cardResults.map((result, index) => {
              return (
                <ReactCardFlip isFlipped={this.isFlipped[index]}>
                  <StyledResultCard key="front">
                    <StyledCard color={"#303F9F"}>
                    </StyledCard>
                  </StyledResultCard>
                  <StyledResultCard key="back">
                    <Tooltip  placement="top" TransitionComponent={Zoom} title={result.userName}>
                      <StyledCard color={result.color}>
                        {result.cardValue}
                      </StyledCard>
                    </Tooltip>
                  </StyledResultCard>
                </ReactCardFlip>
              )
            })
            }
          </CardsWrapper>
        </Wrapper>
      </React.Fragment>
    )
  }
}

decorate(CardResults, {
  isFlipped: observable
});

export {CardResults}
export default inject("store")(observer(CardResults));

