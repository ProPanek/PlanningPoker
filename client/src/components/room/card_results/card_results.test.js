import React from "react"
import { CardResults } from "./card_results"
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

describe("CardResults", () => {
  it("renders without errors", () => {
    const store = {
      room: { waiting : [], cardResults : []}
    }

    const component = shallow(<CardResults store={store} />)
    expect(component).toMatchSnapshot()
  })
})
