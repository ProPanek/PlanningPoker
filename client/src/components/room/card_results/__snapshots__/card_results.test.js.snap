// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardResults renders without errors 1`] = `
<Fragment>
  <Styled(WithStyles(Card))>
    <styled.div />
    <styled.div />
  </Styled(WithStyles(Card))>
</Fragment>
`;
